lib/modules/manager/pub/extract.ts

```diff
@@ -61,14 +76,12 @@ export function extractPackageFile(
     return null;
   }
 
-  const deps = [
-    ...extractFromSection(pubspec, 'dependencies'),
-    ...extractFromSection(pubspec, 'dev_dependencies'),
-    ...extractFlutter(pubspec),
-  ];
-
-  if (deps.length) {
```

Comment:
We're guaranteed to always return at least one dependency because `environment.sdk` (i.e. Dart) is required in the pubspec file, so we would always extract it.